jerry-core/api/jerry.c

```diff
@@ -943,6 +974,274 @@ jerry_value_get_type (const jerry_value_t value) /**< input value to check */
   }
 } /* jerry_value_get_type */
 
+/**
+ * Get the object type of the given value
+ *
+ * @return JERRY_OBJECT_TYPE_NONE - if the given value is not an object
+ *         jerry_object_type_t value - otherwise
+ */
+jerry_object_type_t
+jerry_object_get_type (const jerry_value_t value) /**< input value to check */
+{
+  jerry_assert_api_available ();
+
+  if (!ecma_is_value_object (value))
+  {
+    return JERRY_OBJECT_TYPE_NONE;
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (value);
+  ecma_extended_object_t *ext_obj_p = (ecma_extended_object_t *) obj_p;
+
+  switch (ecma_get_object_type (obj_p))
+  {
+    case ECMA_OBJECT_TYPE_FUNCTION:
+    case ECMA_OBJECT_TYPE_BOUND_FUNCTION:
```

Comment:
As for bound functions, what about returning with the type of the bounded function? It just adds default parameters rather than changing the type.

Reply:
This does exactly the same that you described above. A bound function target must be a **callable** function object, and `JERRY_OBJECT_TYPE_FUNCTION` is returned for all callable function object types.